### PR TITLE
Tabs: port to DS6

### LINKED ISSUE #352: Tabs: create DS6 version
Should be pretty easy to port forward.
DS6 Specs here: https://ebay.invisionapp.com/share/CWN9X1JF6MA#/screens/278353515_Tabs

### PR DESCRIPTION
**Pull request by yomed**
## Description

- create DS6 version of `tabs`
- remove `hijax-tabs`
- minor cleanup

## Context

The `tabs` component has never existed in Skin DS6... until now! A few caveats:
- The DS4 tabs use jquery for the js functionality in the demo page. I did not try to set up the js for the DS6 version in the demo, but I did `link` it locally to eBayUI, and it works fine.
- The design specs call out for more variants and complex behavior like horizontal scrollable tabs. I did not account for that functionality here, and instead focused on getting a working DS6 version up in close parity to the DS4 version.

During some of the cleanup/refactor, I saw that `hijax-tabs` has been commented out of the demo page for a long time, so I simplified the component by removing it.

## References

Closes #352 
Design: https://ebay.invisionapp.com/share/CWN9X1JF6MA#/screens/278353515_Tabs

## Screenshots

screen shot 2018-08-28 at 4 10 11 pm

### CODE REVIEW

**Comment by seangates** [comment]
Crap. Lost my comments!

File: src/less/tab/ds6/tab-mixins.less [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    font-weight: bold;
+}
+
+.hovered-tab-outer() {
+    border-bottom: 2px solid @ds6-color-p002-blue;
+}
+
+.hovered-tab-inner() {
+    color: @ds6-color-p002-blue;
+}
```
Comment (seangates) [inline src/less/tab/ds6/tab-mixins.less]: This mixin doesn't make sense as all the mixins are only used once. Lets normalize them (because the DS4 ones are identical in most cases) and move the rules into the color rules using variables into their respective DS files.

Comment (yomed) [reply src/less/tab/ds6/tab-mixins.less]: I went down that apth, but thought it was clearer this way. Although the mixins are only used once, they  provide clarity. The same mixin names are used across ds4/ds6, but with different mixin definitions, which allows them to share the a lot in `base/tab.less`. I could divide up all the rules to be completely DRY, but I think that the purposes of each section would be muddled with all of them split between several files.

Comment (seangates) [reply src/less/tab/ds6/tab-mixins.less]: **DS4**
```
.tab-outer() {
    align-items: center;
    border-bottom: 2px solid transparent;
    box-sizing: border-box;
    display: flex;
    flex: 1;
    margin: 0 2px;
    min-height: 40px;
    -webkit-tap-highlight-color: @ds4-color-transparent-gray-gainsboro;
}
```
**DS6**
```
.tab-outer() {
    align-items: center;
    border-bottom: 2px solid transparent;
    box-sizing: border-box;
    display: flex;
    flex: 1;
    margin: 0 2px;
    min-height: 40px;
    -webkit-tap-highlight-color: @ds6-color-g204-gray;
}
```

Why can't this go in the base where `.tab-outer()` is used? They are identical except for the last color, and that doesn't need to be a mixin. When DS4 goes away (eventually) we won't need the split files, and the base file will reign.

The pattern we followed for all the other components was not this way, is basically what I'm saying. This will now be different from how we've been doing it.

tl;dr — So, yeah, DRY.

Comment (seangates) [reply src/less/tab/ds6/tab-mixins.less]: >When DS4 goes away (eventually) we won't need the split files, and the base file will reign.

Okay, so we might have a new DS by then. But still ...

Comment (msendlakowski) [reply src/less/tab/ds6/tab-mixins.less]: I see where @yomed is coming from.  Its a little awkward to look across multiple files for declarations that apply to the same selector.  But that's the current pattern everywhere else.  And its DRY, like you said.  If a bug gets fixed in DS6 it could easily get left out of DS4 if we start repeating ourselves.

So I vote for moving these to the base file.

**Comment by msendlakowski** [comment]
I think this is the last new feature we should add to the v1 release though. If we get all bugs fixed and verified we can publish a final release.
